ui/lib/replication/addon/controllers/replication-mode.js

```diff
@@ -1,13 +1,26 @@
 import { alias } from '@ember/object/computed';
 import { inject as service } from '@ember/service';
 import Controller from '@ember/controller';
+import { task, timeout } from 'ember-concurrency';
 
 export default Controller.extend({
   rm: service('replication-mode'),
   replicationMode: alias('rm.mode'),
+  waitForNewClusterToInit: task(function*(replicationMode) {
+    // waiting for the newly enabled cluster to init
+    // this ensures we don't hit a capabilities-self error, called in the model of the mode/index route
+    yield timeout(1000);
+    return this.transitionToRoute('mode', replicationMode);
+  }),
   actions: {
-    onEnable(mode) {
-      return this.transitionToRoute('mode', mode);
+    onEnable(replicationMode, mode) {
+      if (replicationMode == 'dr' && mode === 'secondary') {
+        return this.transitionToRoute('vault.cluster');
+      }
+      if (replicationMode === 'dr') {
+        return this.transitionToRoute('mode', replicationMode);
+      }
+      this.waitForNewClusterToInit.perform(replicationMode);
```

Comment:
so does this mean the `waitForNewClusterToInit` task is hit whenever the mode is `performance`, regardless of the mode being `primary` or `secondary`? i think this could be refactored to make that a little clearer, i.e.:

```
if (dr and seconday) {
 transition somewhere
} else if (dr) {
  transition to a different place
} else {
  wait for the cluster
}
```

Reply:
I like it!